include/envoy/api/io_error.h

```diff
@@ -77,11 +51,21 @@ template <typename T> struct IoCallResult {
 
   virtual ~IoCallResult() {}
 
+  IoCallResult& operator=(IoCallResult&& result) {
+    rc_ = result.rc_;
+    err_ = std::move(result.err_);
+    return *this;
+  }
+
   T rc_;
   IoErrorPtr err_;
 };
 
 using IoCallUintResult = IoCallResult<uint64_t>;
```

Comment:
nit: IOCallUint64Result

Reply:
How about the name for struct IoCallResult then? Isn't it envoy naming style to use camel case?

Reply:
Sorry I meant IoCallUint64Result.

My main point was "Uint" was not clear that you meant 64 bits not 32.

Reply:
switched to IoCallUint64Result